src/main/java/application/SessionController.java

```diff
@@ -166,11 +200,101 @@ public CaseFilterResponseBean filterCasesHQ(@RequestBody CaseFilterRequestBean f
+            for(int i=0; i <options.length; i++){
+                optionsStrings.put(i, options[i].getDisplayText());
+            }
+            MenuResponseBean menuResponseBean = new MenuResponseBean();
+            menuResponseBean.setMenuType(Constants.MENU_MODULE);
+            menuResponseBean.setOptions(optionsStrings);
+            menuResponseBean.setSessionId(menuSession.getSessionId());
+            return menuResponseBean;
+        } else if (nextScreen instanceof EntityScreen){
+            EntityScreen entityScreen = (EntityScreen) nextScreen;
+
+            if (entityScreen.getCurrentScreen() instanceof EntityListSubscreen) {
+                EntityListSubscreen entityListSubscreen = (EntityListSubscreen) entityScreen.getCurrentScreen();
+                String[] rows = entityListSubscreen.getRows();
+                HashMap<Integer, String> optionsStrings = new HashMap<Integer, String>();
+                for(int i=0; i <rows.length; i++){
+                    optionsStrings.put(i, rows[i]);
+                }
+                MenuResponseBean menuResponseBean = new MenuResponseBean();
+                menuResponseBean.setMenuType(Constants.MENU_ENTITY);
+                menuResponseBean.setOptions(optionsStrings);
+                menuResponseBean.setSessionId(menuSession.getSessionId());
+                return menuResponseBean;
+            } else if(entityScreen.getCurrentScreen() instanceof EntityDetailSubscreen){
+
+                EntityDetailSubscreen entityDetailSubscreen = (EntityDetailSubscreen) entityScreen.getCurrentScreen();
+                String[] rows = entityDetailSubscreen.getOptions();
+                HashMap<Integer, String> optionsStrings = new HashMap<Integer, String>();
+                for(int i=0; i <rows.length; i++){
+                    optionsStrings.put(i, rows[i]);
+                }
+                MenuResponseBean menuResponseBean = new MenuResponseBean();
+                menuResponseBean.setMenuType(Constants.MENU_ENTITY);
+                menuResponseBean.setOptions(optionsStrings);
+                menuResponseBean.setSessionId(menuSession.getSessionId());
+                return menuResponseBean;
+            }
+
+        }else if (nextScreen == null){
+            NewSessionResponse response = menuSession.startFormEntry(sessionRepo);
+            String stringResponse = new ObjectMapper().writeValueAsString(response);
+            return response;
+        }
+        return null;
+    }
+
+    @ExceptionHandler(Exception.class)
+    public String handleError(HttpServletRequest req, Exception exception) {
+        log.error("Request: " + req.getRequestURL() + " raised " + exception);
+        exception.printStackTrace();
+
+        JSONObject errorReturn = new JSONObject();
+        errorReturn.put("exception", exception);
+        errorReturn.put("url", req.getRequestURL());
+        errorReturn.put("status", "error");
+        return errorReturn.toString();
```

Comment:
we should also look into email handling so we get errors sent to us: http://docs.spring.io/spring-integration/reference/html/mail.html